node/wfprev-gdb-extractor/server.js

```diff
@@ -0,0 +1,190 @@
+
+    try {
+        // Read GDB and extract coordinates
+        dataset = gdal.open(gdbPath);
+
+        dataset.layers.forEach((layer) => {
+            layer.features.forEach((feature) => {
+                const geom = feature.getGeometry();
+                if (geom) results.push(JSON.parse(geom.toJSON()));
+            });
+        });
+
+        // Important: Close the dataset before cleaning up
+        dataset.close();
+        dataset = null;
+
+        // Send response before cleanup
+        res.json(results);
+
+        // Clean up after response has been sent
+        setTimeout(() => {
+            try {
+                // Use the fs module to delete files one by one instead of rimraf
+                const deleteFolderRecursive = function(directoryPath) {
+                    if (fs.existsSync(directoryPath)) {
+                        fs.readdirSync(directoryPath).forEach((file) => {
+                            const curPath = path.join(directoryPath, file);
+                            if (fs.lstatSync(curPath).isDirectory()) {
+                                // Recursive
+                                deleteFolderRecursive(curPath);
+                            } else {
+                                // Delete file
+                                fs.unlinkSync(curPath);
+                            }
+                        });
+                        fs.rmdirSync(directoryPath);
+                    }
+                };
+
+                try {
+                    // Delete the zip file
+                    if (fs.existsSync(zipPath)) {
+                        fs.unlinkSync(zipPath);
+                    }
+                    
+                    // Delete the extracted directory
+                    deleteFolderRecursive(unzipPath);
+                    
+                    // Ensure uploads directory exists
+                    if (!fs.existsSync("uploads")) {
+                        fs.mkdirSync("uploads");
+                    }
+                } catch (cleanupErr) {
+                    console.error("Manual cleanup error:", cleanupErr);
+                }
+            } catch (error) {
+                console.error("Cleanup operation failed:", error);
+            }
+        }, 2000); // 2 second delay for safer cleanup
+
+    } catch (err) {
+        console.error("Error reading GDB:", err);
+        // Make sure to close the dataset if it was opened
+        if (dataset) {
+            try {
+                dataset.close();
+                dataset = null;
+            } catch (closeErr) {
+                console.error("Error closing dataset:", closeErr);
+            }
+        }
+
+        return res.status(500).send("Failed to read GDB.");
+    }
+}
+
+// TO-DO - update this to use Github secrets in WFPREV-402 terraform tasks
+app.use(cors({
+    origin: ['http://localhost:4200', 'https://wfprev-dev.nrs.gov.bc.ca', 'https://wfprev-tst.nrs.gov.bc.ca/', 'https://wfprev-dev.nrs.gov.bc.ca']
```

Comment:
we have duplicate dev urls here.

Reply:
Yeah It'll be covered in WFPREV-402 but I updated it for the duplicate now